gulpfile.js

```diff
@@ -269,14 +270,34 @@ function resolveInstallerExePath(artifactType) {
   return path.join(zaRoot, artifactName + '-' + pjson.version + artifactType + '-installer.exe');
 }
 
-// download all the installer dependencies so we can package them up into the .exe
+function copyFromPrefetchToDist(currentFile, destFile) {
+  let aPromise = null;
+  if (fs.existsSync(destFile)) { // if file exists in buildFolder, we're done 
+    //console.log('EXISTS <- ' + destFile);
+  } else if (fs.existsSync(currentFile)) { // if file exists in prefetchFolder cache, copy it into buildFolder
+    aPromise = new Promise(function(resolve, reject) {
+      console.log('[INFO] Copy ' + currentFile + ' to ' + destFile);
+      fs.copy(currentFile, destFile, function (err) {
+        if (err) { 
+          return console.error(err); 
+        }
+        //console.log('COPIED <- ' + destFile);
+      })
+    });
+  }
+  return aPromise;
+}
+
+// prefetch all the installer dependencies so we can package them up into the .exe
 gulp.task('prefetch', function(cb) {
-  let counter=0;
+  let promises = [];
+  if (!fs.existsSync(prefetchFolder)) { mkdirp(prefetchFolder, cb); }

```

Comment:
This is the reason why, you call back here and the in promise.all().then().catch();